quinn/src/endpoint.rs

```diff
@@ -432,27 +436,31 @@ pub(crate) struct Shared {
 }
 
 impl State {
-    fn drive_recv<'a>(&'a mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
-        self.recv_limiter.start_cycle();
-        let mut metas = [RecvMeta::default(); BATCH_SIZE];
-        let mut iovs = MaybeUninit::<[IoSliceMut<'a>; BATCH_SIZE]>::uninit();
-        self.recv_buf
-            .chunks_mut(self.recv_buf.len() / BATCH_SIZE)
-            .enumerate()
-            .for_each(|(i, buf)| unsafe {
-                iovs.as_mut_ptr()
-                    .cast::<IoSliceMut>()
-                    .add(i)
-                    .write(IoSliceMut::<'a>::new(buf));
-            });
-        let mut iovs = unsafe { iovs.assume_init() };
+    fn poll_socket(
+        &mut self,
+        primary_socket: bool,
+        cx: &mut Context,
+        now: Instant,
+        iovs: &mut [IoSliceMut; BATCH_SIZE],
+        mut metas: [RecvMeta; BATCH_SIZE],
+    ) -> (
```

Comment:
I'm finding this return type pretty hard to parse. Maybe it should be a three-field `struct` or something?